src/components/CalendarDay.jsx

```diff
@@ -16,7 +16,7 @@ const propTypes = forbidExtraProps({
   day: momentPropTypes.momentObj,
   daySize: nonNegativeInteger,
   isOutsideDay: PropTypes.bool,
-  modifiers: PropTypes.object,
+  modifiers: PropTypes.instanceOf(Set),
```

Comment:
it's worth noting that this may make the PR semver-major, as this adds a requirement for es6-shimmed or above, and breaks IE <= 8 (if it wasn't already broken)